compute/cker/include/cker/train/operation/BinaryArithmetic.h

```diff
@@ -65,6 +66,18 @@ void BinaryArithmeticGrad(const Shape &incoming_shape, const T *incoming_data,
     break;
 
     case ArithmeticType::kMul:
+    {
+      auto const in_map = MapAsVector(incoming_data, incoming_shape);
+      auto const lhs_map = MapAsVector(lhs_data, lhs_shape);
+      auto const rhs_map = MapAsVector(rhs_data, rhs_shape);
+      auto lhs_grad_map = MapAsVector(lhs_grad_data, lhs_grad_shape);
+      auto rhs_grad_map = MapAsVector(rhs_grad_data, rhs_grad_shape);
+
+      lhs_grad_map = in_map.array() * rhs_map.array();
+      rhs_grad_map = in_map.array() * lhs_map.array();
```

Comment:
$L$: LHS
$R$: RHS
$O$: Output (of elementwise division)
$X$: Output (of entire model)

$L \cdot R = O$

$\frac{\partial X}{\partial L} = \frac{\partial O}{\partial L} \cdot \frac{\partial X}{\partial O} = R \cdot \frac{\partial X}{\partial O}$
$\frac{\partial X}{\partial R} = \frac{\partial O}{\partial R} \cdot \frac{\partial X}{\partial O} = L \cdot \frac{\partial X}{\partial O}$